### ISSUE TITLE
Certain devices should not be restored to inventory upon being unequipped

### ISSUE THREAD

**Issue by kostmo** [swarm-game/swarm#918]
Could indicate the ability to be restored to inventory when unequipped with a property called `reusable`.  Or, since maybe it will be more rare for devices to not be re-usable, come up with a word that means the opposite of `reusable` to indicate such.

Conceptually, something like a `compass` should be easy to remove, whereas perhaps `armor plates` would not be.

From a scenario design standpoint, this lets us devise more interesting resource sharing/distribution challenges.

**Comment by byorgey** [comment]
Makes sense.  And I agree that `reusable` will probably be more common than the opposite so it should be the default, and the opposite of `reusable` should be specified as... `disposable`?  `perishable`?  `ephemeral`?

**Comment by kostmo** [comment]
`unrecoverable`?

**Comment by byorgey** [comment]
Sure, any of those work.  Also, are you thinking that `unrecoverable` things *cannot* be `unequip`ped, or just that they disappear if you do?  I could imagine both making sense.

**Comment by kostmo** [comment]
`unrecoverable` == they can be unequipped, but doing so would not return them to the inventory (they disappear).

**Comment by byorgey** [comment]
This should be pretty straightforward to implement, so a great issue for a new contributor to tackle if no one else happens to do it first.  It would just require adding a new `EntityProperty`:

https://github.com/swarm-game/swarm/blob/eb20ea9b0752baaaebe323dd213b21b8bc92ef9c/src/swarm-scenario/Swarm/Game/Entity.hs#L149

and then updating the interpreter code for `Unequip` to check for this property and add the device to the inventory conditionally:

https://github.com/swarm-game/swarm/blob/eb20ea9b0752baaaebe323dd213b21b8bc92ef9c/src/swarm-engine/Swarm/Game/Step/Const.hs#L437-L446

**Comment by byorgey** [comment]
More generally, we could also imagine something like `yields`, where certain entities when unequipped turn into something else.  I'm not sure if that would be worth the extra complication.
